mat64/triangular_test.go

```diff
@@ -315,3 +315,51 @@ func TestTriMul(t *testing.T) {
 	receiver = NewTriDense(3, matrix.Upper, nil)
 	testTwoInput(t, "TriMul", receiver, method, denseComparison, legalTypesUpper, legalSizeTriMul, 1e-14)
 }
+
+func TestTriExp(t *testing.T) {
+	method := func(receiver, a Matrix) {
+		type ExpTrier interface {
+			Exp(a Triangular)
+		}
+		receiver.(ExpTrier).Exp(a.(Triangular))
+	}
+	denseComparison := func(receiver, a *Dense) {
+		receiver.Exp(a)
+	}
+	legalSizeTriMul := func(ar, ac int) bool {
+		// Need both to be square and the sizes to be the same
+		return ar == ac
+	}
+
+	// The legal types are triangles with the same TriKind.
+	// legalTypesTri returns whether both input arguments are Triangular.
+	legalTypes := func(a Matrix) bool {
+		if _, ok := a.(Triangular); !ok {
+			return false
+		}
+		return true
+	}
+	legalTypesLower := func(a Matrix) bool {
+		if !legalTypes(a) {
+			return false
+		}
+		_, kind := a.(Triangular).Triangle()
+		r := kind == matrix.Lower
+		return r
+	}
+	legalTypesUpper := func(a Matrix) bool {
+		if !legalTypes(a) {
+			return false
+		}
+		_, kind := a.(Triangular).Triangle()
+		r := kind == matrix.Upper
+		return r
+	}
+
+	for n := 0; n < 8; n++ {
```

Comment:
Yes, it does, but this code had special cased certain sizes, so it should test explicitly for those sizes. `testOneInput` just tests an arbitrary set of sizes.